app/com/linkedin/drelephant/spark/legacydata/LegacyDataConverters.scala

```diff
@@ -233,6 +233,8 @@ object LegacyDataConverters {
         executorInfo.shuffleRead,
         executorInfo.shuffleWrite,
         executorInfo.maxMem,
+        new Date(executorInfo.addTime),
+        if(executorInfo.removeTime!=null) new Date(executorInfo.removeTime) else null,
```

Comment:
Again , please see if we can handle this with Options

Reply:
Can't change the actual val to option as it has been used extensively. So just used option in the if condition itself.